### PR TITLE
ci: remove unused releases, sync supported platforms with infra-agent

### PR DESCRIPTION
**Pull request by roobre**

### CODE REVIEW

File: .goreleaser.yml [hunk 1]
```diff
@@ -13,20 +13,18 @@ builds:
       - CGO_ENABLED=0
     goos:
       - linux
-      - darwin
```
Comment (paologallinaharbur) [inline .goreleaser.yml:16]: soon we are going to support darwin

File: .goreleaser.yml [hunk 2]
```diff
@@ -42,50 +40,6 @@ builds:
       pre: build/windows/set_exe_properties.sh {{ .Env.TAG }} "prometheus"
 
 nfpms:
```
Comment (paologallinaharbur) [inline .goreleaser.yml:42]: shoudl it be nfpms: {} or remove the line?

File: build/release.mk [hunk 3]
```diff
@@ -45,12 +45,6 @@ release/fix-archive:
 	@echo "===> $(INTEGRATION) === [release/fix-archive] fixing zip archives internal structure"
 	@bash $(CURDIR)/build/windows/fix_archives.sh $(CURDIR)
 
-.PHONY : release/sign/nix
-release/sign/nix:
-	@echo "===> $(INTEGRATION) === [release/sign] signing packages"
-	@bash $(CURDIR)/build/nix/sign.sh
```
Comment (paologallinaharbur) [inline build/release.mk:51]: we can also garbage collect the key from the secrets repo if they are not managed by org

File: .goreleaser.yml [hunk 4]
```diff
@@ -13,20 +13,18 @@ builds:
...
-        goarch: 386
-      - goos: freebsd
-        goarch: arm
-      - goos: freebsd
-        goarch: arm64
+      - 386
+      - mips
+      - mips64
+      - mipsle
+      - mips64le
```
Comment (paologallinaharbur) [inline .goreleaser.yml:24]: do we need this?

Comment (roobre) [reply .goreleaser.yml:24]: No idea, I grabbed those from the agent: https://github.com/newrelic/infrastructure-agent/blob/c9042d152df97c2f87279af557a8afea4494ad6f/build/.goreleaser_linux.yml#L142

File: .goreleaser.yml (hunk 1 above)
Comment (roobre) [reply .goreleaser.yml:16]: Nice point, I'll restore those

File: .goreleaser.yml (hunk 2 above)
Comment (roobre) [reply .goreleaser.yml]: Definitely, good catch.

File: build/release.mk (hunk 3 above)
Comment (roobre) [reply build/release.mk:51]: Omw

**Review by paologallinaharbur — APPROVED** [review]

**Pull request merged**